addons/payment/wizards/onboarding_payment_wizard.py

```diff
@@ -0,0 +1,118 @@
+
+    paypal_checked = fields.Boolean('Paypal', default=lambda self: self._get('paypal_checked'))
+    stripe_checked = fields.Boolean('Credit Card (via Stripe)', default=lambda self: self._get('stripe_checked'))
+    manual_checked = fields.Boolean('Manual Payment', default=lambda self: self._get('manual_checked'))
+
+    paypal_email_account = fields.Char('Paypal Email ID', default=lambda self: self._get('paypal_email_account'))
+    paypal_seller_account = fields.Char('Paypal Merchant ID', default=lambda self: self._get('paypal_seller_account'))
+    paypal_pdt_token = fields.Char('Paypal PDT Token', default=lambda self: self._get('paypal_pdt_token'))
+
+    stripe_secret_key = fields.Char(default=lambda self: self._get('stripe_secret_key'))
+    stripe_publishable_key = fields.Char(default=lambda self: self._get('stripe_publishable_key'))
+
+    manual_name = fields.Char('Method name', default=lambda self: self._get('manual_name'))
+    manual_post_msg = fields.Html('Payment instructions', default=lambda self: self._get('manual_post_msg'))
+
+    def _install_module(self, module_name):
+        module = self.env['ir.module.module'].sudo().search(
+            [('name', '=', module_name)])
+        if module.state not in ('installed', 'to install', 'to upgrade'):
+            module.button_immediate_install()
+
+    @api.multi
+    def add_payment_methods(self):
+        modules_required = ['payment_' + payment_method
+                            for payment_method in self.payment_methods
+                            if getattr(self, payment_method + '_checked')]
+
+        if len(modules_required) > 0 or self['manual_checked'] is True:
+            for module in modules_required:
+                self._install_module(module)
+
+            # if any payment method is selected activate "Online Payment" in invoicing settings
+            self._install_module('account_payment')
+
+            # create a new env including the freshly installed module(s)
+            new_env = api.Environment(self.env.cr, self.env.uid,
+                                      self.env.context)
+
+            # if payment method is checked update the values
+            for payment_method in self.payment_methods:
+                if getattr(self, payment_method + '_checked'):
+                    data = dict((field, self[field]) for field in self.config_fields[payment_method])
+                    new_env.ref('payment.payment_acquirer_' + payment_method).write(data)
+
+            # manual
+            if self.manual_checked:
+                data = dict((field, self['manual_' + field]) for field in self.manual_fields)
+                self._get_manual_payment(new_env).write(data)
+
+            self._mark_step_done()
+
+        return {'type': 'ir.actions.act_window_close'}
+
+    @api.multi
+    def skip_this_step(self):
+        self._mark_step_done()
+        return {'type': 'ir.actions.act_window_close'}
+
+    def _mark_step_done(self):
```

Comment:
Rename into something more accurate, like _action_onboarding_payment_step_done or something like this.